src/enterdb_lib.erl

```diff
@@ -1177,21 +1172,22 @@ make_db_key(KeyDef, HashKeyDef, Key) ->
     {error, Reason::term()}.
 make_db_key([Field | RestD], [Field | RestH], Key, DBKeyList, HashKeyList) ->
     case lists:keyfind(Field, 1, Key) of
-        {_, Val} ->
-            make_db_key(RestD, RestH, Key, [Val|DBKeyList], [Val|HashKeyList]);
+        {Field, Val} ->
+            make_db_key(RestD, RestH, Key, [{Field,Val}|DBKeyList], [{Field,Val}|HashKeyList]);
```

Comment:
What is the effect on stored data size of this change?
Do you use Field string when matching prefix?
You could probably store encoded Field as a binary term like we do for column names (Fields).

Reply:
I haven't measured the data size impact, I should probably do that. Good point.

The idea is to be able to do sext decode directly when reading out data.
So that we don't have to loop over the data again to do decode and add name of fields.
But we could send down field names to C (do decode and add fields) if we go with that optimisation.